### DISCUSSION TITLE
Req File Always undefined

### DISCUSSION THREAD

**Discussion by yassineultimate** [expressjs/multer#1286] [Q&A] [answered]
I have This code to uplad user table  image Profile but when i upload image i get req File  undefined
    
      export const updateuser = async (id:number,  image:any) => {

        const formData = new FormData();
  
    formData.append("image", {
      uri: image,
      name: "uploaded_image.jpg", // Customize the name
      type: "image/jpeg", // Ensure it matches your image type
    });
 
   
    
    const response = await axios.put(`${API_URL}/image/${ id }`, { formData}, {
      headers: {
        "Content-Type": "multipart/form-data",  
      },

    });
     
     return response.data;
     };

in backend

      **********
       const multer = require('multer')
       const storage = multer.memoryStorage();
      const upload = multer({ storage });
       *******
      router.put("/image/:id", upload.single("image"), async (req, res) => {
       try {
         const { id } = req.params; // Image ID to update
         if (!req.file) {
           return res.status(400).send("No file uploaded.");
        }

 
           await User.update({ image: req.file.buffer}, { where: { Id: id } });
            res.sendStatus(200);
          
          
               } catch (error) {
                console.error(error);
         res.sendStatus(500);
      }
     });

**Comment by bhrtsnegi** [comment, marked as answer]
Don't wrap formData in an object. Axios will automatically handle the Content-Type for multipart/form-data when FormData is passed. Also, ensure that image has properties like uri, name, and type. If you're using react-native-image-picker or similar libraries, make sure to extract the proper fields.

**Try This**

`export const updateuser = async (id: number, image: any) => {
  const formData = new FormData();

  formData.append("image", {
    uri: image.uri, // Ensure `image` has a `uri` field
    name: "uploaded_image.jpg", // Set the file name
    type: image.type || "image/jpeg", // Ensure the correct MIME type
  });

  const response = await axios.put(`${API_URL}/image/${id}`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });

  return response.data;
};
`
